app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -189,6 +190,18 @@ class SettingsFragment : PreferenceFragmentCompat() {
             scrollToPreference(args.preferenceToScrollTo)
         }
 
+        isProfilerActive = requireContext().components.core.engine.profiler?.isProfilerActive()
+        if (isProfilerActive!!) {
+            findPreference<Preference>(getPreferenceKey(R.string.pref_key_start_profiler))?.title =
```

Comment:
question: is there a way to refresh this when the dialog closes? I was under the impression `onResume` would take care of that but it doesn't seem to be the case.

We could also simplify this and have a single "Start/stop Profiler` button rather than changing it dynamically.

Reply:
No it doesn't refresh when it closes. the only way would be to use the `navBackStackEntry` as it was done before. I thought it would traverse the tree again, but no. I think we can just add a listener for `lifecycle.ON_RESUME` (which is weird that is being called but not `onResume`)